tree.go

```diff
@@ -55,6 +55,15 @@ func (kl keylist) Swap(i, j int) {
 	kl[i], kl[j] = kl[j], kl[i]
 }
 
+type Stem []byte
+
+func KeyToStem(key []byte) Stem {
+	if len(key) < StemSize {
+		panic(fmt.Errorf("key length (%d) is shorter than expected stem size (%d)", len(key), StemSize))
+	}
```

Comment:
I'm not sure if this is entirely useful, because if this `if` didn't exist we'd get a panic anyway in the next line. Maybe I'd accept it if this returns an error (that shouldn't happen, but at least avoid panicking the process).

@gballet, what's your opinion?

Reply:
sorry I missed this question that seems to be blocking. On the one hand, panicking in production is extremely bad, so an error would be preferred, but at the moment a panic is more useful because it will indicate where this happens. If a stem that is smaller than 31 bytes is used, this is a huge bug that should never hit production. So I'm partial to leaving it as is for the moment.